hub/core/chunk_engine/generator.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+from typing import Generator, Union
+
+from hub.core.chunk_engine.exceptions import ChunkGeneratorError
+
+
+def generate_chunks(
+    content_bytes: bytes,
+    chunk_size: int,
+    last_chunk_num_bytes: Union[None, int] = None,
+) -> Generator[bytearray, None, None]:
```

Comment:
Is there any reason for having `bytes` in and `bytearray` out?

Reply:
yes, because bytes are static & unchanging, but the bytearray is able to be extended (by joining chunks)

Reply:
but i guess i can defer this to the caller too